test/unit/test.xpi.js

```diff
@@ -511,20 +516,66 @@ describe("lib/xpi", function() {
     .catch(done);
   });
 
-  it("Warn when executed against a WebExtension", function(done) {
-    process.chdir(webextensionUnsupportedPath);
+  it("Exclude a manifest.json when a manifest.json file exist", function(done) {
+    process.chdir(webextensionManifestExcludedPath);
 
     // In this test scenario, jpm is executed agains a WebExtension that contains
     // both a package.json and a manifest.json files in the root dir, probably because
     // the add-on is being built from an npm package.
-    var manifest = require(path.join(webextensionUnsupportedPath, "package.json"));
-    xpi(manifest).then(function () {
-      // The XPI file is not supposed to be created successfully.
-      throw Error("Expected error not received");
-    }).catch(function(error) {
-      // The expected error message should point to the web-ext intro on MDN.
-      expect(error.message).to.contains("Getting_started_with_web-ext");
-      done();
-    }).catch(done);
+    var manifest = require(path.join(webextensionManifestExcludedPath, "package.json"));
+    return xpi(manifest)
+      .then(function(xpiPath) {
+        return utils.unzipTo(xpiPath, tmpOutputDir);
+      })
+      .then(function(xpiPath) {
+        var testDoesNotExist = when
+          .all([ "manifest.json" ]
+          .map(function(p) { return path.join(tmpOutputDir, p); })
```

Comment:
this block would be easier for me to read if it was indented to show how it's a method on the Array instance.